lib/ocpp/v16/smart_charging.cpp

```diff
@@ -166,7 +166,7 @@ bool SmartChargingHandler::validate_profile(
     ChargingProfile& profile, const int connector_id, bool ignore_no_transaction, const int profile_max_stack_level,
     const int max_charging_profiles_installed, const int charging_schedule_max_periods,
     const std::vector<ChargingRateUnit>& charging_schedule_allowed_charging_rate_units) {
-    if ((size_t)connector_id > this->connectors.size() or connector_id < 0 or profile.stackLevel < 0 or
+    if ((size_t)connector_id >= this->connectors.size() or connector_id < 0 or profile.stackLevel < 0 or
```

Comment:
In general we should probably use static_cast here instead of using a c-style cast

Reply:
This has been addressed with https://github.com/EVerest/libocpp/pull/929